### PR TITLE
Assorted optimizations

### PR DESCRIPTION
**Pull request by Myocardinal**
I implemented the following changes to decrease the latency of CVVDP inference and backprop. On my machine, forward passes are now 1.5x faster, and forwards+backwards passes are 1.9x faster on full HD images(CPU). Speedup increases with input size. I have a version locally that's significantly faster on GPU (~10x on A100), but contains some refactors that are more convoluted and hurt code legibility.

1. Removed unnecessary allocations (torch.zeros -> torch.empty) for tensor that is completely overwritten.
2. Added separable Gaussian blur.
3. Dropped redundant abs.
4. Factored shared S * ch_gain.
5. Cross-channel masking and colorspace math re-written with einsum.
6. Precompute constant powers (xcm_pow, mask_c_pow, d_max_pow) once in load_config instead.
7. Zero-copy array upload (torch.tensor -> torch.as_tensor).

I validated the change by comparing the new diff's JOD output and gradients against HEAD. Differences are within floating point error.

The SeparableGaussianBlur class can be eliminated from this diff if you add in `kornia` as a dependency. I don't think `torchvision` has a comparable function.

### CODE REVIEW

**Comment by Myocardinal** [comment]
Closed. Opened new PR here: https://github.com/gfxdisp/ColorVideoVDP/pull/55

Needed to do some conflict resolution with dev_0_6_0 branch
